src/components/canvas-tools/image-tool.tsx

```diff
@@ -27,24 +26,45 @@ const defaultImagePlaceholderSize = { width: 200, height: 200 };
 export default class ImageToolComponent extends BaseComponent<IProps, {}> {
 
   public state: IState = { isLoading: true, imageUrl: "assets/image_placeholder.png" };
+  private _asyncRequest: any;
 
   public componentDidMount() {
     const { model: { content } } = this.props;
     const { db } = this.stores;
     const imageContent = content as ImageContentModelType;
-    if (!this.state.hasUpdatedUrl) {
-      fetchImageUrl(imageContent.url, db.firebase, (fullUrl: string) => {
-        getImageDimensions((dimensions: any) => {
-          this.setState({ imageUrl: fullUrl, imageDimensions: dimensions, isLoading: false, hasUpdatedUrl: true });
-        }, undefined, fullUrl);
-      });
+
+    // Migrate Firebase storage relative URLs to full URLs
+    fetchImageUrl(imageContent.url, db.firebase, ((url: string) => {
+      this.handleUpdateImageDimensions(url);
+      if (url !== imageContent.url) {
```

Comment:
I would move this test into the `updateStoredURL()` function since it makes sense for any caller.